corgidrp/l2a_to_l2b.py

```diff
@@ -91,23 +91,55 @@ def flat_division(input_dataset, flat_field):
 
     return flatdiv_dataset
 
-def frame_select(input_dataset, bpix_frac=100., overexp=False, tt_thres=None):
-  
+def frame_select(input_dataset, bpix_frac=1., overexp=False, tt_thres=None):
     """
     
     Selects the frames that we want to use for further processing.
-    Not currently implemented!! 
 
     Args:
         input_dataset (corgidrp.data.Dataset): a dataset of Images (L2a-level)
-        bpix_frac (float): what percent of the image needs to be bad to discard. Default: 100% (not used)
+        bpix_frac (float): greater than fraction of the image needs to be bad to discard. Default: 1.0 (not used)
         overexp (bool): if True, removes frames where the OVEREXP keyword is True. Default: False
         tt_thres (float): maximum allowed tip/tilt in image to be considered good. Default: None (not used) 
 
     Returns:
         corgidrp.data.Dataset: a version of the input dataset with only the frames we want to use
     """
-    return input_dataset.copy()
+    pruned_dataset = input_dataset.copy()
+    select_flags = np.ones(len(input_dataset))
+
+    for i, frame in enumerate(input_dataset.frames):
+        if bpix_frac < 1:
+            numbadpix = np.size(np.where(frame.dq > 0)[0])
```

Comment:
This logic might include pixels that we're comfortable including, e.g. cosmic rays that have been corrected for. I'm not quite sure how to handle it, but maybe we just make this  a future enhancement issue for now. You could imagine e.g. counting the number of dq pixels here that have a specific set of bit values....just thinking out loud.

Reply:
we could have as input a custom bpvalue mask, that species which DQ flags are allowed. Or, we can just hard code it to ignore the DQ = 2 flag for fixed cosmic rays.

Reply:
Yeah, inputting a custom bpvalue mask was basically what I was thinking. I think it would just have to be a single integer. We're currently using DQ=2 to also flag pixels replaced in the Correct bad Pixel step too, so I don't love that option as much.

Reply:
Ok, added new argument to pass in allowed DQ values that the function will ignore